### PR TITLE
Remove border & background from invisible disabled buttons

### PR DESCRIPTION
**Pull request by iansan5653**
Removes the background and border from disabled invisible `Button` and `IconButton` by updating the shared `ButtonBase` CSS styles. It looks like `ButtonBase` is fully shipped to CSS Modules so we don't have to update any styled-system styles for this.

- Closes https://github.com/primer/react/issues/5421

Before:

Demo with enabled & disabled invisible iconbuttons. The disabled button has a border and background.

After (video to show hover/active styling):

https://github.com/user-attachments/assets/fcd28aa1-bd81-4bc0-a0cb-fed90dd4a466

#### Changed

- Removes border & background from `disabled` `invisible` buttons

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/Button/ButtonBase.module.css [hunk 1]
```diff
@@ -430,8 +430,6 @@
     &:disabled,
     &[aria-disabled='true']:not([data-loading='true']) {
       color: var(--button-invisible-fgColor-disabled);
```
Comment (iansan5653) [inline packages/react/src/Button/ButtonBase.module.css:432]: Comparison of rest/disabled colors for reference:

screenshot of enabled/disabled invisible iconbuttons

File: packages/react/src/Button/ButtonBase.module.css [hunk 2]
```diff
@@ -430,8 +430,6 @@
...
-      background-color: var(--button-invisible-bgColor-disabled);
-      border-color: var(--button-invisible-borderColor-disabled);
```
Comment (iansan5653) [inline packages/react/src/Button/ButtonBase.module.css:434]: This change makes these tokens obsolete. Should we delete them entirely? cc @langermank

Comment (langermank) [reply packages/react/src/Button/ButtonBase.module.css:434]: Just took a few min to make a PR for this which we can test and just use this branch to bump versions if you want. 

Could we make a Storybook story just showing all button variants `disabled` in a row, move it to the `dev` stories and add a VRT test for it? 🙌

**Comment by langermank** [comment]
Closing this in favor of https://github.com/primer/react/pull/5440

**Pull request closed without merging**